website/docs/crud.md

```diff
@@ -0,0 +1,259 @@
+```python
+def delete(self, id: int):
+```
+`id` is the id of the column to be deleted.
+
+#### Usage
+```python
+from core_lib.data_layers.data_access.db.crud.crud import CRUD
+from core_lib.data_layers.data_access.db.crud.crud_data_access import CRUDDataAccess
+from core_lib.data_layers.data.db.sqlalchemy.base import Base
+
+from sqlalchemy import Column, Integer, VARCHAR, Boolean
+
+class Customer(Base):
+    __tablename__ = 'customer_data'
+
+    id = Column(Integer, primary_key=True, nullable=False)
+    name = Column(VARCHAR(length=255), nullable=False, default="")
+    email = Column(VARCHAR(length=255), nullable=False, default="")
+    active = Column(Boolean, nullable=False)
+
+
+class CustomerCRUDDataAccess(CRUDDataAccess):
+    def __init__(self):
+        CRUD.__init__(self, Customer, db_handler)
+
+```
+
+
+### CRUDSoftDeleteDataAccess
+Similar to `CRUDDataAccess` but is used to access and handle soft delete.
+For this to work the database object class must extend `SoftDeleteMixin` to create the required columns for soft delete.
+
+```python
+class CRUDSoftDeleteDataAccess(DataAccess, CRUD):
+
+    def __init__(self, db_entity, db: SqlAlchemyDataHandlerRegistry, rule_validator: RuleValidator):
+        CRUD.__init__(self, db_entity, db, rule_validator)
+```
+
+
+#### Functions provided by `CRUDSoftDeleteDataAccess`
+
+- `get(id: int)` overrides the function in `CRUD` class, used to get data from database for a given `id`.
+```python
+def get(self, id: int):
+```
+`id` is the id of the column we want to query.
+
+
+- `delete(id: int)` overrides the function in `CRUD` class, soft deletes the data for the given `id`.
+```python
+def delete(self, id: int):
+```
+`id` is the id of the column to be soft deleted.
+
+#### Usage
+```python
+from core_lib.data_layers.data_access.db.crud.crud import CRUD
+from core_lib.data_layers.data_access.db.crud.crud_soft_data_access import CRUDSoftDeleteDataAccess
+from core_lib.data_layers.data.db.sqlalchemy.base import Base
+from core_lib.data_layers.data.db.sqlalchemy.mixins.soft_delete_mixin import SoftDeleteMixin
+
+from sqlalchemy import Column, Integer, VARCHAR, Boolean
+
+class Customer(Base, SoftDeleteMixin):
+    __tablename__ = 'customer_data'
+
+    id = Column(Integer, primary_key=True, nullable=False)
+    name = Column(VARCHAR(length=255), nullable=False, default="")
+    email = Column(VARCHAR(length=255), nullable=False, default="")
+    active = Column(Boolean, nullable=False)
+
+
+class CustomerCRUDSoftDeleteDataAccess(CRUDSoftDeleteDataAccess):
+
+    def __init__(self):
+        CRUD.__init__(self, Customer, db_handler)
+
+```
+
+
+### CRUDSoftDeleteWithTokenDataAccess
+This class is similar to `CRUDSoftDeleteDataAccess` but is uses `SoftDeleteMixin` as well as `SoftDeleteTokenMixin` which
+creates the `delete_token` column, if someone wants to index the deleted columns then this class will be used, as 

```

Comment:
`for the reason that indexing on a DateTime column is slow, adding the `delete_token` will allow as to index the delete control column